### DISCUSSION TITLE
How to make sure a string is a view-string and let phpstan know it?

### DISCUSSION THREAD

**Discussion by Bbansjkl** [larastan/larastan#1461] [Q&A]
In my current project we have a few views that are served by creating a view-string dynamically, like this:  

```php
view('mails.'.$order->getDynamicViewParam().'.topic.template');
```  

To which phpstan/larastan (rightfully) complains that a view-string is expected, not a string.  

But, how do I tell phpstan that this is a view-string? I tried concatenating the string before passing it and checking it like the [Custom Types doc](https://github.com/nunomaduro/larastan/blob/master/docs/custom-types.md) mentions 

> Any `string` that passes the `view()->exists($string)` test is also a valid `view-string`.

```php
$viewString = 'mails.'.$order->getDynamicViewParam().'.topic.template';
if (!view()->exists($viewString)) {
  throw new Exception("View not found: {$viewString}")
}
$content = view($viewString);
```
But that doesn't seem to convince phpstan.

I could of course just declare it via `/** @phpstan-var view-string $viewString */`, but that defeats the point, as it wouldn't be much better than just telling phpstan to ignore the line - and my IDE (PhpStorm) also doesn't recognize `view-string` and shows me an additional warning about that.

Another thing I could do is creating a global function that takes a `string` as param, puts it through the test and returns a `view-string`, but that seems wasteful - and runs again into the issue of my IDE not recognizing `view-string`.

So, is there any way to ensure phpstan/larastan knows that a string passes the `view()->exists()` test?

**Comment by mad-briller** [comment]
if this is merged you can use `view()->exists($viewString)` to check it is a view-string:
https://github.com/nunomaduro/larastan/pull/1464
